src/front/spv/mod.rs

```diff
@@ -1984,6 +1990,72 @@ impl<I: Iterator<Item = u32>> Parser<I> {
                 Op::Fwidth | Op::FwidthFine | Op::FwidthCoarse => {
                     self.parse_expr_derivative(expressions, crate::DerivativeAxis::Width)?;
                 }
+                Op::ArrayLength => {
+                    inst.expect(5)?;
+                    let result_type_id = self.next()?;
+                    let result_id = self.next()?;
+                    let structure_id = self.next()?;
+                    let member_index = self.next()?;
+
+                    // Validate that all the types are correct.
```

Comment:
I don't think we need this, necessarily. If type expectations don't match, Validator should yell.
And if we want to check the result type ID, we'd probably do it in a more generic way for all expressions.